Framework/API/src/IFunction.cpp

```diff
@@ -61,13 +62,34 @@ struct TieNode {
            other.right.end();
   }
 };
+const std::vector<std::string> EXCLUDEUSAGE = {"CompositeFunction"};
 } // namespace
+/**
+ * Constructor
+ */
+IFunction ::IFunction()
+    : m_isParallel(false), m_handler(nullptr), m_chiSquared(0.0) {}
 
 /**
  * Destructor
  */
 IFunction::~IFunction() { m_attrs.clear(); }
 
+/**
+Registers the usage of the function with the UsageService
+ */
+void IFunction::registerFunctionUsage() {
+  if (!Kernel::UsageService::Instance().isEnabled()) {
+    return;
+  }
+  if (std::find(EXCLUDEUSAGE.cbegin(), EXCLUDEUSAGE.cend(), name()) ==
+          EXCLUDEUSAGE.cend() &&
+      !m_isRegistered) {
+    m_isRegistered = true;
+    Kernel::UsageService::Instance().registerFeatureUsage(
+        Kernel::FeatureType::Function, name(), true);
```

Comment:
So it's the final argument here that controls whether the table gets internal set to true or not. To be consistent with algorithms can we pass through from `Fit` the `isChild` value?